WordPress/gutenberg
# Focus pattern inserter search when activating zoom out inserter

## What?


Focuses the Pattern inserter search when an inserter is clicked in Zoom Out mode.

Closes https://github.com/WordPress/gutenberg/issues/63867

## Why?


Currently when clicking on `+` inserters in Zoom Out mode nothing happens. You just get an insertion point show up but nothing else. The idea in https://github.com/WordPress/gutenberg/pull/63896 is that the insertion point expands so it's obvious that you use drag and drop but this won't work for keyboard users.

Therefore we need a means for keyboard users to be able to insert patterns into the canvas.

## How?


Exposes a `ref` which is programmatically focused when an inserter is activated in Zoom Out mode.

Because Zoom Out is part of the `core/block-editor` package and the search input is part of `core/editor` package we cannot select from the `editor` store inside the Zoom Out inserter components. Nor is it feasible to pass a prop down to the Zoom Out inserters.

Therefore this PR utilises the block editor settings to expose the ref from the editor package to the block editor package code.

## Testing Instructions





- New Page
- Enter Zoom Out mode.
- Add some patterns
- Click on inserter in the Zoom Out canvas.
- See the patterns inserter search receives focus.

### Testing Instructions for Keyboard


Repeat the steps above but use the keyboard and ensure that you can access the patterns inserter.

## Screenshots or screencast 

https://github.com/user-attachments/assets/ae6b6628-7086-4e52-9b0a-183e2c87e984
